src/Disks/ObjectStorages/DiskObjectStorage.cpp

```diff
@@ -583,6 +623,12 @@ void DiskObjectStorage::applyNewSettings(
     }
 
     IDisk::applyNewSettings(config, context_, config_prefix, disk_map);
+    applyNewSettingsImpl(config, config_prefix);
+}
+
+void DiskObjectStorage::applyNewSettingsImpl(const Poco::Util::AbstractConfiguration & config, const String & config_prefix)
+{
+    enable_page_cache = config.getBool(config_prefix + ".enable_page_cache", false);
```

Comment:
btw why is it a server config setting? seems no problem to have it is a profile setting and it seems useful to be able to change it per query.

Update: I see now you have a `force_enable_page_cache` setting for that purpose. But may be it is enough to have a single profile setting then?